grammar.js

```diff
@@ -39,72 +43,111 @@ module.exports = grammar({
       $.union_shape_statement,
       $.operation_shape_statement,
       $.resource_shape_statement,
-      $.service_shape_statement
+      $.service_shape_statement,
+      $.enum_shape_statement
     ),
 
     absolute_root_shape_id: $ => seq($.namespace, '#', $.identifier),
     root_shape_id: $ => choice($.absolute_root_shape_id, $.identifier),
     shape_id_member: $ => seq('$', $.identifier),
-    shape_id: $ => seq($.root_shape_id, repeat($.shape_id_member)),
+    shape_id: $ => prec.left(seq($.root_shape_id, repeat($.shape_id_member))),
     control_var_name: $ => seq('$', $.node_object_key, ':'),
     namespace: $ => seq($.identifier, repeat(seq('.', $.identifier))),
 
-    simple_shape_statement: $ => seq($.simple_type_name, $.identifier),
-    list_shape_statement: $ => seq('list', $.identifier, $.shape_members),
-    set_shape_statement: $ => seq('set', $.identifier, $.shape_members),
-    map_shape_statement: $ => seq('map', $.identifier, $.shape_members),
-    structure_shape_statement: $ => seq('structure', $.identifier, $.shape_members),
-    union_shape_statement: $ => seq('union', $.identifier, $.shape_members),
+    simple_shape_statement: $ => seq($.simple_type_name, $.identifier, optional($.mixins)),
+    list_shape_statement: $ => seq('list', $.identifier, optional($.mixins), $.shape_members),
+    set_shape_statement: $ => seq('set', $.identifier, optional($.mixins), $.shape_members),
+    map_shape_statement: $ => seq('map', $.identifier, optional($.mixins), $.shape_members),
+    structure_shape_statement: $ => seq(
+      'structure',
+      $.identifier,
+      optional($.structure_resource),
+      optional($.mixins),
+      $.shape_members
+    ),
+    union_shape_statement: $ => seq('union', $.identifier, optional($.mixins), $.shape_members),
 
-    service_shape_statement: $ => seq('service', $.identifier, $.node_object),
-    operation_shape_statement: $ => seq('operation', $.identifier, $.node_object),
-    resource_shape_statement: $ => seq('resource', $.identifier, $.node_object),
+    service_shape_statement: $ => seq('service', $.identifier, optional($.mixins), $.node_object),
+    operation_shape_statement: $ => seq('operation', $.identifier, optional($.mixins), $.operation_body),
+    resource_shape_statement: $ => seq('resource', $.identifier, optional($.mixins), $.node_object),
```

Comment:
- Add support for the mixins syntax
- Add support for the structure resource syntax
- Update the operation shape body to support inline structure definitions.